### PR TITLE
Fix estimate gas RPC failing for clique when no blocks have been created

### PR DESCRIPTION
**Pull request by jframe**
## PR description
Fixes issue when using Clique, if the chain height is still at 0 then the estimateGas rpc fails with a internal server error.

This change special cases block 0 so we return address 0 for the proposer in this case instead of failing with an IllegalArgumentException as the genesis block does not have proposer seals.

### CODE REVIEW

File: consensus/clique/src/main/java/tech/pegasys/pantheon/consensus/clique/CliqueBlockHashing.java [hunk 1]
```diff
@@ -23,6 +23,7 @@
 import java.util.function.Supplier;
 
 public class CliqueBlockHashing {
+  private static final Address ADDRESS_ZERO = Address.extract(Hash.ZERO);
```
Comment (ajsutton) [inline consensus/clique/src/main/java/tech/pegasys/pantheon/consensus/clique/CliqueBlockHashing.java]: I suspect we should just introduce `Address.ZERO = Address.fromHexString('0x0')` in the `Address` class.

File: consensus/clique/src/main/java/tech/pegasys/pantheon/consensus/clique/CliqueBlockHashing.java [hunk 2]
```diff
@@ -48,6 +49,9 @@ public static Hash calculateDataHashForProposerSeal(
    */
   public static Address recoverProposerAddress(
       final BlockHeader header, final CliqueExtraData cliqueExtraData) {
+    if (header.getNumber() == 0) {
```
Comment (ajsutton) [inline consensus/clique/src/main/java/tech/pegasys/pantheon/consensus/clique/CliqueBlockHashing.java]:
```suggestion
    if (header.getNumber() == BlockHeader.GENESIS_BLOCK_NUMBER) {
```

Please don't ever hard code 0 as the genesis block number. One day we're going to have to support variable genesis block numbers.

File: consensus/clique/src/main/java/tech/pegasys/pantheon/consensus/clique/CliqueBlockHashing.java (hunk 1 above)
Comment (jframe) [reply consensus/clique/src/main/java/tech/pegasys/pantheon/consensus/clique/CliqueBlockHashing.java]: done

File: consensus/clique/src/main/java/tech/pegasys/pantheon/consensus/clique/CliqueBlockHashing.java (hunk 2 above)
Comment (jframe) [reply consensus/clique/src/main/java/tech/pegasys/pantheon/consensus/clique/CliqueBlockHashing.java]: done

File: consensus/clique/src/main/java/tech/pegasys/pantheon/consensus/clique/CliqueBlockHashing.java [hunk 3]
```diff
@@ -48,6 +47,9 @@ public static Hash calculateDataHashForProposerSeal(
    */
   public static Address recoverProposerAddress(
       final BlockHeader header, final CliqueExtraData cliqueExtraData) {
+    if (header.getNumber() == BlockHeader.GENESIS_BLOCK_NUMBER) {
```
Comment (rain-on) [inline consensus/clique/src/main/java/tech/pegasys/pantheon/consensus/clique/CliqueBlockHashing.java]: nit: Personally, I'd move this inside the "isPresent" check - just to minimise the conditionals executed on every check ... its probably a minor benefit at best.

Comment (jframe) [reply consensus/clique/src/main/java/tech/pegasys/pantheon/consensus/clique/CliqueBlockHashing.java]: done

**Pull request merged**